File: articles/self-control-commitment-devices-behavioral-economics/go/commitment_choice.go
package main

import (
	"fmt"
	"math"
)

func discountedFutureValue(beta, delta, benefit float64, periods int) float64 {
	return beta * math.Pow(delta, float64(periods)) * benefit
}

func chooseCommitment(beta, delta, futureBenefit, immediateTemptation, commitmentCost float64, periods int) bool {
	patientValue := discountedFutureValue(beta, delta, futureBenefit, periods)
	temptationValue := immediateTemptation - commitmentCost
	return patientValue >= temptationValue
}

func main() {
	fmt.Printf("Synthetic patient choice under commitment: %v\n", chooseCommitment(0.72, 0.97, 1000.0, 600.0, 300.0, 12))
}

File: articles/self-control-commitment-devices-behavioral-economics/c/commitment_choice.c
#include <stdio.h>
#include <math.h>

double discounted_future_value(double beta, double delta, double benefit, int periods) {
    return beta * pow(delta, periods) * benefit;
}

int choose_commitment(double beta, double delta, double future_benefit, double immediate_temptation, double commitment_cost, int periods) {
    double patient_value = discounted_future_value(beta, delta, future_benefit, periods);
    double temptation_value = immediate_temptation - commitment_cost;
    return patient_value >= temptation_value;
}

int main(void) {
    int patient_choice = choose_commitment(0.72, 0.97, 1000.0, 600.0, 300.0, 12);
    printf("Synthetic patient choice under commitment: %d\n", patient_choice);
    return 0;
}
